designsafe/middleware.py

```diff
@@ -33,6 +34,29 @@ def process_request(self, request):
                                       'or <a href="https://www.mozilla.org/en-US/firefox/new/">Firefox</a> '
                                       'if you experience issues.')
 
+class DesignsafeProfileUpdateMiddleware:
+
+    """
+    Middleware to check if a user's profile has the update_required flag set to 
+    True, and force them to update their profile if so.
+    """
+
+    def process_request(self, request):
+        blocked_path = request.path.startswith('/data')\
+            or request.path.startswith('/applications')\
+            or request.path.startswith('/rw/workspace')\
+            or request.path.startswith('/recon-portal')\
+            or request.path.startswith('/dashboard')
+        if request.user.is_authenticated:
```

Comment:
Consider falttening the nested ifs for less cyclomatic complexity:
```suggestion
        if request.user.is_authenticated and request.user.profile.update_required and blocked_path:
            # show warning
        elif request.user.is_authenticaated:
            return None
```